x-pack/legacy/plugins/maps/public/embeddable/map_embeddable.js

```diff
@@ -152,7 +152,20 @@ export class MapEmbeddable extends Embeddable {
       );
     }
 
-    this._store.dispatch(replaceLayerList(this._layerList));
+    // Use hiddenLayers input to set visibility of layers
+    let layerList = this._layerList;
+
+    if (this.input.hiddenLayers) {
```

Comment:
Can you please move this logic into an action handler in https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/maps/public/actions/map_actions.js. Then dispatch the action after this._store.dispatch(replaceLayerList(layerList)).